File: backend/quick_wallet/database/models/storage/card.py
import enum

from sqlalchemy import Column, Enum, ForeignKey
from sqlalchemy.dialects.postgresql import TEXT, UUID

from quick_wallet.database.models.base import BaseTable

from .shop import ShopCategoryEnum


class CardTypeEnum(enum.Enum):
    STANDARD = "STANDARD"
    UNKNOWN = "UNKNOWN"


class CardColorEnum(enum.Enum):
    PINK = "ffc0cb"
    TURQUOISE = "30d5c8"


class Card(BaseTable):
    __tablename__ = "card"

    type = Column(Enum(CardTypeEnum), nullable=False, doc="Card type")
    owner_id = Column(
        UUID(as_uuid=True),
        ForeignKey("user.id", onupdate="CASCADE", ondelete="CASCADE"),
        nullable=False,
        doc="Card owner ID from User table",
    )
    barcode_data = Column(
        TEXT,
        nullable=False,
        doc="Card barcode data",
    )
    shop_id = Column(UUID(as_uuid=True), nullable=True, unique=False, doc="Shop ID from shop table")
    image_url = Column(
        TEXT,
        nullable=True,
        unique=False,
        doc="URL to load card image (if type=STANDARD)",
    )
    shop_name = Column(
        TEXT,
        nullable=False,
        unique=False,
        doc="Name of the shop which was written by user or from shop table (if type=STANDARD)",
    )
    color = Column(
        Enum(CardColorEnum),
        nullable=True,
        unique=False,
        doc="Card color in mobile interface (if type=UNKNOWN)",
    )
    map_search_string = Column(
        TEXT,
        nullable=True,
        unique=False,
        doc="String for map api to find the nearest shop",
    )
    category = Column(Enum(ShopCategoryEnum), nullable=False, doc="Card type")

File: mobile/app/src/main/java/com/example/quickwallet/domain/model/Card.kt
package com.example.quickwallet.domain.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize


enum class CardType {
    STANDARD,
    UNKNOWN
}

@Parcelize
data class Card(
    val id: String,
    val type: CardType,
    val barcodeData: String,
    val shopName: String,
    val imageUrl: String?,
    val color: String?,
    val category: String
): Parcelable

val cards = listOf(
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    ),
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    ),
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    ),
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    ),
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    ),
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    ),
    Card(
        "",
        CardType.STANDARD,
        "",
        "",
        "Пятерочка",
        "https://www.tadviser.ru/images/e/ed/%D0%9F%D1%8F%D1%82%D1%91%D1%80%D0%BE%D1%87%D0%BA%D0%B0_LOGO.jpg",
        ""
    )
)